modifications/dataclass.cpp

```diff
@@ -0,0 +1,135 @@
+#include "dataclass.h"
+
+DataClass::DataClass(QObject *parent) : QObject(parent)
+{
+    this->generateToken();
+}
+
+bool DataClass::droneIDChanged(QString vehicleID){
+
+    if(drone.serialId  == vehicleID)
+    {
+        return  false;
+    }
+    else
+    {
+        drone.serialId = vehicleID;
+        return true;
+    }
+}
+
+//TOKEN REQUEST
+void DataClass::generateToken()
+{
+    QNetworkRequest request = QNetworkRequest(QUrl("https://id.openskies.sh/oauth/token/"));
```

Comment:
Refactor this URL into a config param from some common config file

Reply:
Like we discussed, there should be a JSON / YAML config file put at a specific location that should hold this information. Just FYI, Flight Passport (https://id.openskies.sh) is a OAUTH server and OAUTH is a standard so you can use any server so the URL must be made configurable.

Reply:
Use parameter read from config XML here

Reply:
Done